Parsedown.php

```diff
@@ -1210,33 +1195,17 @@ protected function inlineText($text)
     {
         $Inline = array(
             'extent' => strlen($text),
-            'element' => array(
-                'elements' => array(),
-            ),
+            'element' => array(),
         );
 
-        if ($this->breaksEnabled)
-        {
-            $Inline['element']['elements'] = self::pregReplaceElements(
-                '/[ ]*\n/',
-                array(
-                    array('name' => 'br'),
-                    array('text' => "\n"),
-                ),
-                $text
-            );
-        }
-        else
-        {
-            $Inline['element']['elements'] = self::pregReplaceElements(
-                '/(?:[ ][ ]+|[ ]*\\\\)\n/',
-                array(
-                    array('name' => 'br'),
-                    array('text' => "\n"),
-                ),
-                $text
-            );
-        }
+        $safeText = self::escape($text, true);
+
+        $Inline['element']['rawHtml'] = preg_replace(
+            $this->breaksEnabled ? '/[ ]*+\n/' : '/(?:[ ]*+\\\\|[ ]{2,}+)\n/',
+            "<br />\n",
+            $safeText
+        );
+        $Inline['element']['allowRawHtmlInSafeMode'] = true;
```

Comment:
I'm still in two minds about this change – it's quite a bit faster because plaintext is pretty common and it saves us creating several arrays for each chunk of it, however it blinds any AST post-processor to the content structure. It also perhaps risks encouraging use of `rawHtml` + `allowRawHtmlInSafeMode` over the AST, which is less than ideal where it can be trivially avoided.

Reply:
I'm not so sure whether breaks are *structural* elements or rather *styling* elements... Paragraphs are structural elements without any doubt, but single line breaks? Furthermore, isn't this a great example of AST processing adding unnecessary complexity? I don't think that we encourage people to use `rawHtml` + `allowRawHtmlInSafeMode` - it's rather a great example of how and under which circumstances it can and should be used. We shouldn't condemn `allowRawHtmlInSafeMode` :wink:

Reply:
> I'm not so sure whether breaks are *structural* elements or rather *styling* elements... 

Yes, you're correct in that they're used for style. I was more going for "structure" here to mean that breaks represent a different kind of object in the syntax tree than plaintext does. I don't really have much concern for *why* the elements are used (rather that they are elements) :p

I definitely don't want to condemn `allowRawHtmlInSafeMode`, but if someone needs to create something simple like `<p>foobar</p>`, then I'd much rather they wrote it like:

```php
$Element = array('name' => 'p', 'text' => 'foobar');
```

than

```php
$Element = array('rawHtml' => '<p>foobar</p>', 'allowRawHtmlInSafeMode' => true);
```

Especially so when `foobar` becomes a variable.

Even if someone does all the escaping correctly, it's much easier to not to have to if it isn't required.

Perhaps an example of how to use raw HTML in a safe way like this is valuable though.

Reply:
> I definitely don't want to condemn allowRawHtmlInSafeMode, but if someone needs to create something simple like `<p>foobar</p>`, then I'd much rather …

> Perhaps an example of how to use raw HTML in a safe way like this is valuable though.

Full ACK :+1: